# Lista 05 - IC + Teste de Hipóteses

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import math

from scipy.stats import binom

# Exercício 1

Utilize o dataset dados-covid-mg para explorar a seguinte hipótese:

Hipótese: Em MG cerca de 50% da população já foi detectada com COVID.

Dataset obtido no openDataSUS, um recorte do estado de MG. Para facilitar, algumas colunas foram removidas e o atributo classificacaoFinal foi binarizado.

In [2]:
df = pd.read_csv('dados-covid-mg.csv')
df.head()

id profissionalSaude tipoTeste resultadoTeste        estado  \
0  R3y2Di9K0s               Não    RT-PCR       Negativo  MINAS GERAIS   
1  4awz0HPaCl               Sim    RT-PCR       Negativo  MINAS GERAIS   
2  0RjrlH0bs5               Não       NaN            NaN  MINAS GERAIS   
3  83B4gJDLpU               Não       NaN            NaN  MINAS GERAIS   
4  H8GKXacI4T               Não       NaN            NaN  MINAS GERAIS   

              municipio  classificacaoFinal  
0              Varginha                   0  
1   Carmópolis de Minas                   0  
2  Conselheiro Lafaiete                   0  
3               Mariana                   0  
4              Januária                   0

A) Calcule a prevalência de infectados na amostra, armazene na variável p.

In [9]:
# YOUR CODE HERE
p = len(df[df['classificacaoFinal']==1])/len(df)
p
#raise NotImplementedError()

0.409622523888832

B) Calcule a variância (utilizando p), armazene em var.

In [10]:
# YOUR CODE HERE
var = np.var(df['classificacaoFinal'])
var
#raise NotImplementedError()

0.24183191181177524

C) Calcule o intervalo de confiança dessa estimativa, considerando 95% de chance. O resultado deve ser uma tupla a,b armazenada em ic.

In [12]:
# YOUR CODE HERE
inferior = p - 1.96*math.sqrt(var)/math.sqrt(len(df))
superior = p + 1.96*math.sqrt(var)/math.sqrt(len(df))
ic = (inferior, superior)
ic
#raise NotImplementedError()

(0.40754359272076335, 0.4117014550569007)

D) Quais as afirmações analíticas podem ser feitas com base nessa exploração? Se julgar necessário plote gráficos que corroborem a análise.

p nos diz que 40.96% dos pacientes na amostra estão infectados.

O intervalo de confiança nos diz que, caso fossem feitas outras amostragens parecidas, 95% delas teriam uma quantidade de contaminados variando entre 40.75% e 41.17%

Assim, podemos concluir que a quantidade de contaminados na população brasileira muito provavelmente também gira em torno de 40.96%, podendo variar entre 40.75% e 41.17%

# Exercício 2

Hipótese Nula: A prevalência de COVID para profissionais de saúde é igual a de não profissionais de saúde.

Utilize o teste de permutação para randomizando os dados, aproximar o p-valor e validar se a hipótese é aceita.

In [ ]:
#Convertendo as series para numpy
profissionais = df['profissionalSaude'].to_numpy()
classificacaoFinal = df['classificacaoFinal'].to_numpy()

A) Calcule t_obs, armazene em t_obs.

In [ ]:
# YOUR CODE HERE
raise NotImplementedError()

B) Implemente o teste de permutação para os dados

Relembrando: Shuffle, computar diferença média, fazer isso n vezes.

Para sua análise utilize o valor de N que considerar adequado, no entanto na entrega fixe N em 1000. Armazene a lista de diferenças em diferencas.

In [ ]:
# YOUR CODE HERE
raise NotImplementedError()

C) Plote o histograma e avalie qual a chance de um t_obs maior que os dados.

In [ ]:
# YOUR CODE HERE
raise NotImplementedError()

D) A hipótese nula é aceita ou rejeitada? Explique

In [ ]:
# YOUR CODE HERE
raise NotImplementedError()